### ISSUE TITLE
javadoc gradle task is failing due to "error: no tag name after"

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#7089]
This failure to compile the javadoc is causing some build failures.

this is easy to reproduce:

```
~/projects/logstash (git)-[master] % gradle javadoc
:logstash-core:compileJava UP-TO-DATE
:logstash-core:processResources UP-TO-DATE
:logstash-core:classes UP-TO-DATE
:logstash-core:javadoc
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/DeadLetterQueueFactory.java:47: error: no tag name after @
     * Retrieves an existing {@link DeadLetterQueueWriter} associated with {@param id}, or
                                                                           ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/DeadLetterQueueFactory.java:53: error: no tag name after @
     *                for each {@param id}
                               ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java:58: warning: no description for @param
     * @param queuePath
       ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java:59: warning: no description for @param
     * @param maxSegmentSize
       ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java:60: warning: no description for @throws
     * @throws IOException
       ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java:62: warning: no @param for maxQueueSize
    public DeadLetterQueueWriter(Path queuePath, long maxSegmentSize, long maxQueueSize) throws IOException {
           ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java:96: warning: no description for @throws
     * @throws IOException
       ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/RecordIOReader.java:146: warning: no description for @return
     * @return
       ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/RecordIOReader.java:211: warning: no description for @return
     * @return
       ^
/Users/joaoduarte/projects/logstash/logstash-core/src/main/java/org/logstash/common/io/RecordIOReader.java:212: warning: no description for @throws
     * @throws IOException
       ^
2 errors
8 warnings
:logstash-core:javadoc FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':logstash-core:javadoc'.
> Javadoc generation failed. Generated Javadoc options file (useful for troubleshooting): '/Users/joaoduarte/projects/logstash/logstash-core/build/tmp/javadoc/javadoc.options'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

```

regardless of fixing the specific reason the doc is failing to compile, it may be worth disabling javadoc task completely as the javadocs aren't used.

**Comment by andrewvc** [comment]
We also need to call `gradle assemble` as a test task in travis.

**Comment by andrewvc** [comment]
Fixed in #7108
